readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -70,6 +70,11 @@ function attach_elastic_search_query_sphinx(data) {
                         }
 
                         var link = result.path + "?highlight=" + $.urlencode(query);
+                        // If we aren't on the main domain of the subproject, link to it.
+                        // TODO: we should always redirect to the main domain instead.
+                        if (result.path.startsWith('/projects/') && !window.location.href.startsWith(result.domain)) {
```

Comment:
I think on .com we allow users to customize this path, so we probably need something smarter here in the future, but this should work for 99% of users for now.

Reply:
yeah, I think redirecting to the main project should fix all this, but that's a bigger change in proxito